### PR TITLE
VTOL: Pusher assist: add configuration for enabling it in LAND /disable below some altitude

### PR DESCRIPTION
**Pull request by sfuhrer**
Enables the pusher assist feature also in landing mode.

Adds new parameter VT_FWD_TRUST_EN to set when pusher assist should be active (e.g. turn off if below MPC_ALT_LAND1).

**Describe problem solved by this pull request**
This PR solves two current limitations:
- the pusher assist (resp. tilt forward for tiltrotor VTOL) feature currently is disabled in LAND mode (to increase safety in case the pusher/puller doesn't have enough ground clearance. This is usually bad if landing in high head winds, as then the vehicle needs to tilt forward to fight it.
- if the pusher/puller of a VTOL doesn't have enough ground clearance and the user lands in position or altitude mode the propeller can touch the ground

**Describe your solution**
Add new parameter to set when pusher assist should be active:
-completely disabled (default)
-enabled in pos, alt and auto mode (except LAND)
-enabled in pos, alt and auto mode if above MPC_LAND_ALT1
-enabled in pos, alt and auto mode if above MPC_LAND_ALT2
-enabled in pos, alt and auto mode

**Test data / coverage**
SITL and flight tested.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@sfuhrer Do you think you could add a docs PR for this - ie perhaps update http://docs.px4.io/master/en/flight_modes/land.html#vtol to explain how land mode works/has changed by this?

**Comment by sfuhrer** [comment]
> @sfuhrer Do you think you could add a docs PR for this - ie perhaps update http://docs.px4.io/master/en/flight_modes/land.html#vtol to explain how land mode works/has changed by this?

@hamishwillee yes I can do that once this is in. We should also document the pusher assist somewhere, don't think it is.

File: src/modules/vtol_att_control/vtol_type.cpp [hunk 1]
```diff
@@ -422,6 +422,43 @@ bool VtolType::is_channel_set(const int channel, const int target)
 
 float VtolType::pusher_assist()
 {
+	// Altitude above ground is distance sensor altitude if available, otherwise local z-position
+	float dist_to_ground = -_local_pos->z;
+
+	if (_local_pos->dist_bottom_valid) {
```
Comment (sfuhrer) [inline src/modules/vtol_att_control/vtol_type.cpp:428]: @RomanBapst I assume dist_bottom_valid is only true when you have a distance sensor connected? How come then that it still has a value even without one, shouldn't it then be NAN?

**Comment by hamishwillee** [comment]
> yes I can do that once this is in. We should also document the pusher assist somewhere, don't think it is.

Awesome. Yes, pusher-assist docs would be great too.

**Pull request merged**